### ISSUE TITLE
Download JoinSplit Groth16 verification parameters and load them from disk

### ISSUE THREAD

**Issue by dconnolly** [ZcashFoundation/zebra#3037]
## Motivation

In order to verify Sprout JoinSplit-on-Groth16 proofs, we need to load the public verification parameters, which are several hundred MB in size.

### Designs

- [ ] If not cached, download parameter files in parallel with sync
    - https://github.com/zcash/zcash/blob/49ffee3f20b972dc3aa75e422c67523251cf088b/zcutil/fetch-params.sh
- [ ] Read from disk location
- [ ] Check the hash to match
- [ ] Turn into JoinSplit on Groth16 VerificationKey type
- [ ] Instantiate static instance of groth16::Verifier with these parameters

### Design Questions

Where do we store the parameters?
  - Zebra already has a cache directory for its database

How do we avoid loading them in tests that don't need them?

How do we test this code without downloading 500 MB?

## Related Work

https://github.com/ZcashFoundation/zebra/issues/3038

**Comment by dconnolly** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Engineering-Team-5fb24d9264a3e8000e666a9e/issues/ZcashFoundation/zebra/3037?planning-poker) with ZenHub @conradoplg @jvff @mpguerra @oxarbitrage @upbqdn

**Comment by str4d** [comment]
> Where do we store the parameters?

If you look for / use the same folder as `zcashd` then you may be able to avoid downloading anything at all: `~/.zcash-params`. Likewise, if you download to there, then `zcashd` will also Just Work without any downloading.

I had grand plans to also enable `zcashd` to find the parameters at other predictable locations (so e.g. providing the parameters as a separate Debian package for Debian installs, that could be shared between projects), but never got around to it; we could pick this back up.

**Comment by str4d** [comment]
The issue I was thinking of is https://github.com/zcash/zcash/issues/3147

**Comment by teor2345** [comment]
> > Where do we store the parameters?
> 
> If you look for / use the same folder as `zcashd` then you may be able to avoid downloading anything at all: `~/.zcash-params`. Likewise, if you download to there, then `zcashd` will also Just Work without any downloading.

This is the solution I settled on in PR #3057.

> I had grand plans to also enable `zcashd` to find the parameters at other predictable locations (so e.g. providing the parameters as a separate Debian package for Debian installs, that could be shared between projects), but never got around to it; we could pick this back up.

This is out of scope for our current sprints - we're focused on validating Sprout before NU5 mainnet activation.
